### PR TITLE
syn: intrinsic_functions.h with main

### PR DESCRIPTION
**Pull request by Pranavchiku**
Towards #4700

### CODE REVIEW

File: src/libasr/pass/intrinsic_functions.h [hunk 1]
```diff
@@ -763,9 +763,9 @@ namespace Abs {
                 std::to_string(input_kind) + " output kind: " + std::to_string(output_kind),
                 loc, diagnostics);
         } else {
-            ASRUtils::require_impl(ASRUtils::check_equal_type(input_type, output_type, true),
-                "The input and output type of elemental intrinsics must exactly match, input type: " +
-                input_type_str + " output type: " + output_type_str, loc, diagnostics);
+            // ASRUtils::require_impl(ASRUtils::check_equal_type(input_type, output_type, true),
+            //     "The input and output type of elemental intrinsics must exactly match, input type: " +
+            //     input_type_str + " output type: " + output_type_str, loc, diagnostics);
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:768]: Why don't we need this check? If the `input_type` is not complex, we are now not checking anything. Is that correct?

**Comment by certik** [comment]
Otherwise it's fine, but the above seems like a change we don't want.

File: src/libasr/pass/intrinsic_functions.h (hunk 1 above)
Comment (gxyd) [reply src/libasr/pass/intrinsic_functions.h:768]: I'll check this locally to see if we need this condition in *simplifier_pass* branch or not, if we don't I'll send a PR to remove this from *simplifier_pass* branch and then undo it from here.

Comment (gxyd) [reply src/libasr/pass/intrinsic_functions.h:768]: The CI for PR here: https://github.com/lfortran/lfortran/pull/5132 uncommenting this change, it passes CI.

**Review by gxyd — APPROVED** [review]
LGTM!

File: src/libasr/pass/intrinsic_functions.h (hunk 1 above)
Comment (certik) [reply src/libasr/pass/intrinsic_functions.h]: Excellent!

**Comment by certik** [comment]
Good.

**Pull request merged**